worldedit-core/src/main/java/com/sk89q/worldedit/command/BrushCommands.java

```diff
@@ -325,6 +337,62 @@ public void butcherBrush(Player player, LocalSession session,
+                           @Arg(desc = "The name of the image")
+                               String imageName,
+                           @Arg(desc = "The size of the brush", def = "5")
+                               double radius,
+                           @Arg(desc = "The intensity of the brush", def = "5")
+                               double intensity,
+                           @Switch(name = 'e', desc = "Erase blocks instead of filling them")
+                               boolean erase,
+                           @Switch(name = 'f', desc = "Don't change blocks above the selected height")
+                               boolean flatten,
+                           @Switch(name = 'r', desc = "Randomizes the brush's height slightly.")
+                               boolean randomize) throws WorldEditException {
+        Optional<AssetLoader<ImageHeightmap>> loader = worldEdit.getAssetLoaders().getAssetLoader(ImageHeightmap.class, imageName);
+
+        if (loader.isPresent()) {
+            worldEdit.checkMaxBrushRadius(radius);
+            BrushTool tool = session.getBrushTool(player.getItemInHand(HandSide.MAIN_HAND).getType());
+
+            ImageHeightmapLoadTask task = new ImageHeightmapLoadTask(loader.get(), imageName);
+            AsyncCommandBuilder.wrap(task, player)
+                .registerWithSupervisor(worldEdit.getSupervisor(), "Loading asset " + imageName)
+                .setDelayMessage(TranslatableComponent.of("worldedit.asset.load.loading"))
+                .setWorkingMessage(TranslatableComponent.of("worldedit.asset.load.still-loading"))
+                .onSuccess(TranslatableComponent.of("worldedit.brush.image.equip", TextComponent.of((int) radius)), heightmap -> {
+                    tool.setSize(radius);
+                    tool.setBrush(new ImageHeightmapBrush(heightmap, intensity, erase, flatten, randomize), "worldedit.brush.image");
+                })
+                .onFailure(TranslatableComponent.of("worldedit.asset.load.failed"), worldEdit.getPlatformManager().getPlatformCommandManager().getExceptionConverter())
+                .buildAndExec(worldEdit.getExecutorService());
+        } else {
+            AsyncCommandBuilder.wrap((Callable<Component>) () -> {
+                List<Path> imageNames = worldEdit.getAssetLoaders().getFilesForAsset(ImageHeightmap.class);
+                TextComponent.Builder builder = TextComponent.builder();
+                int i = 0;
+                for (Path path : imageNames) {
+                    builder.append(TextComponent.of(MoreFiles.getNameWithoutExtension(path), i % 2 == 0 ? TextColor.GRAY : TextColor.WHITE));
+                    if (i <= imageNames.size()) {
+                        builder.append(TextComponent.of(", "));
+                    }
+                    i++;
+                }
+                return TranslatableComponent.of("worldedit.brush.image.unknown", TextComponent.of(imageName), builder.build());
```

Comment:
Shouldn't this be a paginated box of some sort? I don't really like the idea of dumping a potentially massive list into chat.